src/library/Waveform.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This is the library for creating the waveform. 
+ * It requires the ID of canvas element where the waveform
+ * will be rendered as initialization input. When the waveform
+ * is created, it requires the analyserNode as real-time audio 
+ * input data for drawing the correct waveform.
+ * 
+ * There is no library dependence, this entire waveform is
+ * created based on the canvas element.
+ */
+
+class Waveform {
+
+  /**
+   * @constructor This is the constructor for initializing waveform
+   *   Since width and height is initialized in contructor, the
+   *   waveform's width and height can not be changed after initialization.
+   * @param {String} canvasId  An ID of a canvas element where
+   *   the waveform will be rendered.
+   * @param {AnalyserNode} analyserNode The analysis node
+   *   which connect with the audio context.
+   * @param {Number} fftSize The window size in samples that is used 
+   *   when performing a Fast Fourier Transform (FFT) to get frequency
+   *   domain data in the AnalyserNode.
+   */
+  constructor(canvasId, analyserNode, fftSize) {
+    /** @private @const {!canvas} Selected waveform canvas element */
+    this.canvas_ = document.querySelector(canvasId);
+    /** @private @const {!canvasContext} Canvas context */
+    this.canvasContext_ = this.canvas_.getContext('2d');
+    /** @private @const {!width} Selected waveform canvas width */
+    this.width_ = this.canvas_.width;
+    /** @private @const {!height} Selected waveform canvas height */
+    this.height_ = this.canvas_.height;
+    /** @private @const {!currentX} Current X axis in waveform */
+    this.currentX_ = 0;
+    /** @private @const {!previousY} Previous Y axis in waveform */
+    this.previousY_ = this.height_ / 2;
+    /** @private @const {!analyser} AnalyserNode of audio context */
+    this.analyser_ = analyserNode;
+    this.analyser_.fftSize = fftSize;
+    /** @private @const {!bufferLength} Total number of data 
+     *     points avliable to AudioContext sampleRate. */
+    this.bufferLength_ = this.analyser_.frequencyBinCount;
+    /** @private @const {!dataArray} AnalyserNode of audio context */
+    this.dataArray_ = new Float32Array(this.bufferLength_);
+  }
+
+  /**
+   * This is the draw function for creating waveform.
```

Comment:
This comment is about "what this method is".

What we want for the comment is usually "why". In this case, I'd talk about how this function gets executed and the frequency of it. For example, this class doesn't have a timer or a render loop; it requires an external driver for the actual rendering.